src/htm/utils/Log.hpp

```diff
@@ -28,32 +28,36 @@
 
 namespace htm {
 enum class LogLevel { LogLevel_None = 0, LogLevel_Minimal=1, LogLevel_Normal=2, LogLevel_Verbose=3 };
-static LogLevel NTA_LOG_LEVEL = LogLevel::LogLevel_Minimal; // change this in your class to set log level
+// change this in your class to set log level using Network.setLogLevel(level);
+extern thread_local LogLevel NTA_LOG_LEVEL; 
```

Comment:
I guess non NetworkAPI classes can also use the static Network.setLogLevel(), but writing directly to this variable should work too, right?

Reply:
> but writing directly to this variable should work too, right?

yes.  you can just assign to it.  `NTA_LOG_LEVEL = LogLevel::LogLevel_Verbose;`

We could use some other class to put the setLogLevel(  ) function on that is accessible by the algorithms but I did not find anything general enough.  Anyway, the python binding interface has it on the Network class.